src/tests/ftest/util/daos_io_conf.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import os
+
+from apricot import TestWithServers
+from command_utils import ExecutableCommand
+from test_utils import TestPool
+
+class IoConfGen(ExecutableCommand):
+    """Base IoConf execute command class.
+
+    :avocado: recursive
+    """
+    def __init__(self, command, path="", env=None):
+        """Create a ExecutableCommand object.
+
+        Uses Avocado's utils.process module to run a command str provided.
+
+        Args:
+            command (str): string of the command to be executed.
+            path (str, optional): path to location of command binary file.
+                Defaults to "".
+        """
+        super(IoConfGen, self).__init__("", command, path)
+        self.verbose = True
+        self.env = env
+
```

Comment:
I think what both @saurabhtandan and I are trying to convey is the inheritance of `ExecutableCommand` by `IoConfGen` is not being fully utilized.  

There are really three main features of the `ExecutableCommand` class:
1) It produces a command string including optional arguments when those arguments are specified
2) It provides a simplified one-method-call approach for reading these optional arguments from the test yaml by defining the yaml parameter key names and a fixed path in which to search for these keys
3) It provides a `run()` method for executing the command as a foreground or background process

All of these three features combine to define an object that is portable for use by other developers without needing to know too much about the command - they can just create the object, call `get_params()`, and call `run()`.

Reply:
How do I pass the different argument to run_gen_io_conf() from the code? not from the yaml file

Reply:
So assuming in the `__init__()` there is the following (and it passes `command="daos_gen_io_conf"` in the `super.__init__()`):
```
self.rank_size = FormattedParameter("-g {}")
self.target_size = FormattedParameter("-t {}")
self.record_size = FormattedParameter("-s {}", 3)
```

Then you can do the following:
```
io_conf = IoConfGen("/path/to/exe")
print(str(io_conf))
io_conf.rank_size.value = 10
print(str(io_conf))
io_conf.target_size.value = 5
print(str(io_conf))
```

The output from the print will look like this.  Note that the first `print()` only includes a `-s` command line argument because it is the only FormattedParameter object defined with a default.
```
/path/to/exe/daos_gen_io_conf -s 3
/path/to/exe/daos_gen_io_conf -g 10 -s 3
/path/to/exe/daos_gen_io_conf -g 10 -s 3 -t 5
```

Reply:
Updated the code to use FormattedParameter class.